core/src/test/scala/org/bykn/bosatsu/NormalizationTest.scala

```diff
@@ -178,7 +176,7 @@ out=match Pair(1, "two"):
 
 """
       ), "Match/Structs",
-      Struct(0,List(Literal(Integer(BigInteger.valueOf(3))), Literal(Str("two")), Literal(Integer(BigInteger.valueOf(1)))))
+      Struct(0,List(Literal(Lit(3)), Literal(Lit(1)), Literal(Lit(1))))
```

Comment:
this looks really suspect to me. I looks like the `two` has been replaced by `1`... the tests pass though.

@snoble can you take a look at this? If pattern matching were broken, I would assume at least one of the evalueation tests or the existing bosatsu code that is tested would hit it.

Reply:
Weird, it's the constructor code that seems to be causing the issue. Getting closer

Reply:
Interesting, the constructor is fine. It's the reduction of `App(Lambda`

Reply:
Something weird is happening with a bunch of variables named `a` and `b` showing up in the environment. I don't know where they are coming from yet

Reply:
Ohhhh I see... I'm handling shadowing wrong of lambda variables. I'm seeing an environment like this 
`env: (Map(Name(a) -> ExpressionKeyTag(LambdaVar(2),Set()), Name(b) -> ExpressionKeyTag(LambdaVar(0),Set())),List(Name(b), Name(a), Name(a)))`
But `Name(a)` should point to `LambdaVar(1)` not `LambdaVar(2)` in the map.

I just need to remember where the `zip` that does that is and fix that